packages/core/src/factory.ts

```diff
@@ -71,25 +72,51 @@ export function factory<Resource, Input, Output, Config extends IPrismConfig>(
 
       return pipe(
         TaskEither.fromEither(components.route({ resources, input })),
-        TaskEither.chain(resource => inputValidation(resource, input, config)),
-        TaskEither.chain(({ resource, inputValidations }) => mockOrForward(resource, input, config, inputValidations)),
-        TaskEither.map(({ output, resource, inputValidations }) => {
-          const outputValidations = config.validateResponse
-            ? pipe(
-                Option.fromEither(Either.swap(components.validateOutput({ resource, element: output }))),
-                Option.getOrElse<IPrismDiagnostic[]>(() => [])
-              )
-            : [];
+        TaskEither.fold(
+          error => {
+            if (!config.errors && isProxyConfig(config)) {
```

Comment:
I have a feeling that the code got complicated. I believe adding more separation between proxying and mocking would improve a lot here.

Below is kind of "work in progress" however, it shows clearly the idea of how to simplify the flow:
```
       return pipe(
        isProxyConfig(config)
          ? pipe(
            TaskEither.fromEither(components.route({ resources, input })),
            TaskEither.chain(resource => inputValidation(resource, input, config)),
            TaskEither.alt(error => config.errors ? TaskEither.left(error) : TaskEither.right({ inputValidations: [
                {
                  message:
                    "The selected route hasn't been found and the errors is set false. Prims will proxy the request to the upstream server but no validation will happen",
                  severity: DiagnosticSeverity.Warning,
                },
              ] })),
            TaskEither.chain(({ inputValidations }) => forward(input, config, inputValidations)),
          )
        : pipe(
            TaskEither.fromEither(components.route({ resources, input })),
            TaskEither.chain(resource => inputValidation(resource, input, config)),
            TaskEither.chain(({ resource, inputValidations }) => mock(resource, input, config, inputValidations)),
          ),
        TaskEither.map(outputValidation)
      );
```

This requires separating `forward` and `mock` and ofc some polishing around typing (I'll be happy to do that if you want).

Reply:
Yeah I agree the flow is getting complicated — I'm willing to try your solution. I'll get back on this and see if that will bring some improvements.

Reply:
Ok I have tried and the resulting flow isn't really that simplified. I would temporarily keep it in this way but if you're willing to give it a spin — be my guest! Just open a PR starting from this brach and we'll go through together!